onnxruntime/core/providers/cpu/generator/constant_of_shape_base.h

```diff
@@ -78,8 +78,7 @@ class ConstantOfShapeBase {
     auto* t_proto_p = t_proto.get();
 #endif
     if (info.GetAttr<ONNX_NAMESPACE::TensorProto>("value", t_proto_p).IsOK()) {
-      ORT_ENFORCE(t_proto_p->dims_size() == 1, "Must have a single dimension");
-      ORT_ENFORCE(t_proto_p->dims()[0] == 1, "Must have a single dimension of 1");
+      ORT_ENFORCE(utils::GetTensorShapeFromTensorProto(t_proto_p).Size(), "The value attribute of ConstantOfShape must be a single-element tensor");
```

Comment:
The size must be equal to 1.
Also I would actually adjust the title of the PR.

Reply:
Sorry thanks - for some reason I missed the comparison when editing